controller/pkg/controlplane/controlplane.go

```diff
@@ -21,17 +21,5 @@ type DefaultsArgs struct {
 
 // SpecDeepEqual returns true if the two ControlPlaneOptions are equal.
 func SpecDeepEqual(spec1, spec2 *gwtypes.ControlPlaneOptions, envVarsToIgnore ...string) bool {
-	if !reflect.DeepEqual(spec1.DataPlane, spec2.DataPlane) {
-		return false
-	}
-
-	if !reflect.DeepEqual(spec1.Extensions, spec2.Extensions) {
-		return false
-	}
-
-	if !reflect.DeepEqual(spec1.WatchNamespaces, spec2.WatchNamespaces) {
-		return false
-	}
-
-	return true
+	return reflect.DeepEqual(spec1, spec2)
```

Comment:
As I understand, you keep the parameter `env Vars Ignore ...string` so as not to break an API. It'll be clean up later, won't be?

So I recommend do now `envVarsToIgnore ...string` to `_ ...string`